initialization/initialize.go

```diff
@@ -199,42 +199,73 @@ func (init *Initializer) initFile(id []byte, fileIndex int, labelGroupsPerFile u
 		fileIndex, infileParallelism)
 
 	numOfWorkers := infileParallelism
-	workersOutputChans := make([]chan []byte, numOfWorkers)
+	workersOutputChans := make([]chan [][]byte, numOfWorkers)
 	errChan := make(chan error, 0)
 	finishedChan := make(chan struct{}, 0)
+	batchSize := 100
+	chanBuffer := 100
 
+	// CPU workers.
 	fileOffset := uint64(fileIndex) * labelGroupsPerFile
 	for i := 0; i < numOfWorkers; i++ {
 		i := i
-		workersOutputChans[i] = make(chan []byte, 1000) // TODO: make chan buffer configurable
-		offset := i
+		workersOutputChans[i] = make(chan [][]byte, chanBuffer)
+		workerOffset := i
 		go func() {
-			// Calculate labels in groups, write them to disk
-			// and append them as leaves in the merkle tree.
-			for position := uint64(offset); position < labelGroupsPerFile; position += uint64(numOfWorkers) {
-				lg := CalcLabelGroup(id, position+fileOffset, Difficulty(init.cfg.Difficulty))
-				workersOutputChans[i] <- lg
+			// Calculate labels in groups and write them to channel in batches.
+			iterator := 0
+			position := uint64(workerOffset)
+			batch := make([][]byte, batchSize)
+			for position < labelGroupsPerFile {
```

Comment:
I think it is better if the batches will contain consecutive labels instead of labels in a round robin manner. It will simplify the IO part a lot

Reply:
I originally started with this approach but it seemed to me even more complicated. I'll give it another try.